### PR TITLE
[ISSUE-461] Support Spark 3.3

### PR DESCRIPTION
**Pull request by kaijchen**
### What changes were proposed in this pull request?

1. Add maven profile `spark3.3`
2. Enable tests for Spark 3.3 in CI
3. Fix broken tests

### Why are the changes needed?

#461 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

https://github.com/kaijchen/incubator-uniffle/actions/runs/3879507232

### CODE REVIEW

File: .github/workflows/parallel.yml [hunk 1]
```diff
@@ -62,6 +62,7 @@ jobs:
           - spark3
           - spark3.2
           - spark3.2.0
+          - spark3.3
```
[the remark this answers is not in the document]
Comment (kaijchen) [reply .github/workflows/parallel.yml:65]: `spark3` is actually Spark 3.1.
`spark3.2.0` is listed apart from `spark3.2` due to some accidental API change.

We should change is in another PR.

**Review by jerqi — APPROVED** [review]
LGTM

**Review by zuston — APPROVED** [review]
lgtm +1

**Comment by kaijchen** [comment]
Thanks @jerqi and @zuston for the review.

**Pull request merged**